### DISCUSSION TITLE
I don't know why but my edges are forming away from my mouse pointer.

### DISCUSSION THREAD

**Discussion by surya0180** [xyflow/xyflow#3300] [Q&A]
When I click on any handle and drag the edge, the edge is forming away from my mouse cursor. Please refer to the GIF attached below. I know what is causing this issue, but I want to fix this from reactflow itself instead of fixing the one that is causing this issue. Basically I am zooming my webpage a bit depending on the device innerHeight and innerWidth. 

Here is the what I am doing

```
document.documentElement.style.zoom = width / 1400.
```

Where `width` is the total width of the device that is viewing my website and `1400` is the pixel length that I want it to scale for. When this is in use, the reactflow edges are behaving like this 

<img src="https://github.com/wbkd/react-flow/assets/56592197/c34b058e-17b4-42b4-b9fe-e6d1f534745a" alt="drawing" width="500"/>

Any idea how I can fix this in reactflow itself?

**Comment by moklick** [comment]
`document.documentElement.style.zoom = width / 1400` this is not what you should do. You can set the zoom level by using [`defaultViewport`](https://reactflow.dev/docs/api/react-flow-props/#defaultviewport) or [`setViewport`](https://reactflow.dev/docs/api/react-flow-instance/#setviewport)

**Reply by surya0180** [reply]
```
document.documentElement.style.zoom = width / 1400
```

is not set specifically for reactflow component. It is set for the whole application in App.js. I haven't changed any zoom properties in reactflow component.

**Reply by moklick** [reply]
yep, that's the issue here. You can't scale the reactflow container and I don't think that it is a good idea in general to zoom a whole app.

**Reply by surya0180** [reply]
So can I fix this issue somehow without removing the zooming logic? I cannot remove the zooming logic as I am not authorized to do so 😓.

**Reply by moklick** [reply]
I am not sure if it's possible to invert the zoom just on the react flow container somehow. You should ask someone if you could change it, since `style.zoom` is not recommended to use at all 

> "This feature is non-standard and is not on a standards track. Do not use it on production sites facing the Web: it will not work for every user. There may also be large incompatibilities between implementations and the behavior may change in the future."

https://developer.mozilla.org/en-US/docs/Web/CSS/zoom

**Reply by surya0180** [reply]
That's the thing I can't remove it even if its not a good practise to use zoom. When you mentioned `I am not sure if it's possible to invert the zoom just on the react flow container somehow` I got an idea to use useEffect and revert the zoom to normal like this

```
useEffect(() => {
    document.documentElement.style.zoom = 1;
    return () => {
      var zoomLevel = width / 1400;
      document.documentElement.style.zoom = zoomLevel;
    };
  }, []);
```

This worked like a charm, but I expected the fix from reactflow component. Nevertheless, thanks for hinting that invert idea.

**Reply by moklick** [reply]
haha what? :D
